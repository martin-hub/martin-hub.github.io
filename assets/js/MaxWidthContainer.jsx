import React from 'react';
import styled from 'styled-components';
import {media} from './media.jsx';

/**
 * This component provides varying horizontal margin responsively
 */
const MaxWidthContainer = styled.div`
  margin: 0 auto;
  width: 100%;

  ${media.xxlarge`
    max-width: 1700px;
  `}
  
  ${media.xlarge`
    max-width: 1400px;
  `}
  
  ${media.large`
    max-width: 1100px;
  `}
`;

export default MaxWidthContainer;
