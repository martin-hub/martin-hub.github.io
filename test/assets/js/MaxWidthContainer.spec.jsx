import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import MaxWidthContainer from 'MaxWidthContainer.jsx';

describe("MaxWidthContainer", () => {
  test('renders correctly', () => {
    expect(renderer.create(<MaxWidthContainer />).toJSON()).toMatchSnapshot();
  });
});