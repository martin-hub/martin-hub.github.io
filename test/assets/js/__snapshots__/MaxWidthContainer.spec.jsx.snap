// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaxWidthContainer renders correctly 1`] = `
.c0 {
  margin: 0 auto;
  width: 100%;
}

@media (min-width:120.0625em) {
  .c0 {
    max-width: 1700px;
  }
}

@media (max-width:120em) {
  .c0 {
    max-width: 1400px;
  }
}

@media (max-width:90em) {
  .c0 {
    max-width: 1100px;
  }
}

<div
  className="c0"
/>
`;
